File: backend/app/schemas/alert.py
"""Alert data contracts."""

from datetime import datetime

from pydantic import Field

from app.models.alert import AlertSeverity, AlertStatus, AlertType
from app.schemas._operational import OperationalSchema


class AlertSchema(OperationalSchema):
    id: str | None = None
    type: AlertType
    severity: AlertSeverity
    status: AlertStatus = AlertStatus.OPEN
    track_id: str | None = None
    sensor_id: str | None = None
    reason: str = Field(min_length=1, max_length=512)
    metadata: dict = Field(default_factory=dict)
    created_at: datetime | None = None
    updated_at: datetime | None = None
    acknowledged_at: datetime | None = None
    resolved_at: datetime | None = None

File: apps/operator/src/types/alert.ts
export type AlertType =
  | 'TRACK_DETECTED'
  | 'TRACK_LOST'
  | 'UNKNOWN_TRACK'
  | 'GEOFENCE_BREACH'
  | 'SENSOR_OFFLINE'
  | 'SENSOR_DEGRADED'
  | 'DATA_QUALITY_LOW';

export type AlertSeverity = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
export type AlertStatus = 'OPEN' | 'ACKNOWLEDGED' | 'RESOLVED';

export interface Alert {
  id: string;
  type: AlertType;
  severity: AlertSeverity;
  status: AlertStatus;
  track_id?: string | null;
  sensor_id?: string | null;
  reason: string;
  metadata?: Record<string, unknown>;
  created_at: string;
  updated_at: string;
  acknowledged_at?: string | null;
  resolved_at?: string | null;
}

export interface AlertListResponse {
  items: Alert[];
  total: number;
  limit: number;
  offset: number;
}
